packages/yourloops/pages/hcp/team-member-add-dialog.tsx

```diff
@@ -0,0 +1,134 @@
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import * as React from "react";
+
+import { makeStyles, Theme } from "@material-ui/core/styles";
+
+import Button from "@material-ui/core/Button";
+import Checkbox from '@material-ui/core/Checkbox';
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import TextField from "@material-ui/core/TextField";
+
+import { TeamMemberRole } from "../../models/team";
+import { t } from "../../lib/language";
+import { REGEX_EMAIL } from "../../lib/utils";
+
+import { AddMemberDialogContentProps } from "./types";
+
+interface AddMemberDialogProps {
+  addMember: null | AddMemberDialogContentProps;
+}
+
+const dialogClasses = makeStyles((theme: Theme) => {
+  return {
+    dialogContent: {
+      display: "flex",
+      flexDirection: "column",
+      minWidth: "25em",
+    },
+    buttonCancel: {
+      marginRight: theme.spacing(2),
+    },
+  };
+});
+
+function AddMemberDialog(props: AddMemberDialogProps): JSX.Element | null {
+  const { addMember } = props;
+
+  const [email, setEMail] = React.useState("");
+  const [role, setRole] = React.useState("viewer" as TeamMemberRole);
+  const [buttonDisabled, setButtonDisabled] = React.useState(true);
+  const classes = dialogClasses();
+
+  let teamName = "";
+  if (addMember !== null) {
+    teamName = addMember.team.name;
+  }
+
+  const handleChangeEMail = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
+    const eMail = e.target.value;
+    setEMail(eMail);
+    setButtonDisabled(!REGEX_EMAIL.test(eMail));
+  };
+  const handleChangeRole = (_e: React.ChangeEvent<HTMLInputElement>, checked: boolean): void => {
+    setRole(checked ? "admin" : "viewer");
```

Comment:
Maybe we should have an enum for the different roles ?

Reply:
Yes good idea.
They have now implemented it, I didn't check the new stuff available in the language.